packages/platforms/delta/x86-64/x86-64-delta-ag8032/onlp/builds/src/module/src/tlvinfo.c

```diff
@@ -0,0 +1,796 @@
+/*
```

Comment:
Can you explain how you're using this TVL code? The onlplib already provides all ONIE TLV parsing and the upper ONLP layer already accepts your TVL eeprom blob for parsing.

Reply:
this TVL code is use to check the eeprom content and make sure whether it is a invalid TVL in the eeprom, which is not influence the ONLP layer  parsing

Reply:
All TLV processing a validation should be done using the common code already present in the existing support library: https://github.com/opencomputeproject/OpenNetworkLinux/blob/a2ef3b938040bef5f33b1923d5e827da2b61f73b/packages/base/any/onlp/src/onlplib/module/inc/onlplib/onie.h

If this requires additional features I would prefer we augment the common ONIE TLV processing to support it rather than reimplement it here.

Reply:
the TVL code  is upper function based on the TVL common code to find the fan or psu mode and series information and that is a platform code which is not the additional features .